pkg/logs/writer_test.go

```diff
@@ -0,0 +1,243 @@
+package logs
+
+import (
+	"bytes"
+	"testing"
+
+	pb "github.com/tektoncd/results/proto/v1alpha2/results_go_proto"
+)
+
+type mockGetLogServer struct {
+	receivedData *bytes.Buffer
+}
+
+func (m *mockGetLogServer) Send(log *pb.Log) error {
+	if m.receivedData == nil {
+		m.receivedData = &bytes.Buffer{}
+	}
+	_, err := m.receivedData.Write(log.GetData())
+	return err
+}
+
+// Test case: "Hello!" has got 6 bytes. Chunks size is 5. Remain is 6 - 5 = 1.
+// NewBufferedWriter.Write([]byte) should send 5 bytes(1 chunk) to sender
+// and save 1 byte in the buffer.
+// NewBufferedWriter.WriteRemain() should send last byte to sender.
+func TestBufferedWriterWhenContentIsLongerThenChunk(t *testing.T) {
```

Comment:
Consider making these unit tests [table-driven](https://dave.cheney.net/2019/05/07/prefer-table-driven-tests), since the validations are similar.

Reply:
Changed to table to driven.